### DISCUSSION TITLE
Question about Cell center

### DISCUSSION THREAD

**Discussion by nahidrahmati** [UvaCsl/HemoCell#19] [General discussion]
Hello, 
I want to calculate the Cell center (whole center volumetric center), is there any property  in Hemocell that is related to the whole particle center position?

**Comment by nahidrahmati** [comment]
In addition, I want to know if there is any reference for the repulsive force equation function which is calculated based on r_const*(1/(distance/r_cutoff)).

**Comment by MaxvdKolk** [comment]
Dear @nahidrahmati,

> I want to calculate the Cell center (whole center volumetric center), is there any property in Hemocell that is related to the whole particle center position?

To obtain a cell's centre position, you can use the `CellInformationFunctionals::calculateCellPosition` function. An example of it's usage can be found in the `examples/oneCellShear` case, see [here](https://github.com/UvaCsl/HemoCell/blob/4dbcb0b2bfe9e36cbe9f778d6a54e787a9457c4b/examples/oneCellShear/oneCellShear.cpp#L143) and [here](https://github.com/UvaCsl/HemoCell/blob/4dbcb0b2bfe9e36cbe9f778d6a54e787a9457c4b/examples/oneCellShear/oneCellShear.cpp#L149). This populates an array (`hemo::Array<T,3>`) with the x, y, z positions in lattice units. If you would like the position in micron, make sure to insert the unit conversion by division with `1e-6/param::dx` (similar to the `oneCellShear` example). For multiple cells, you can iterate over the `info_per_cell` map to find each position.

> In addition, I want to know if there is any reference for the repulsive force equation function which is calculated based on r_const*(1/(distance/r_cutoff)).

I am unsure if there is a HemoCell specific publication regarding the repulsion force and its cut-off distance. Maybe @gzavo could provide some details. As far as I see, this currently considers only an inversely proportional force with respect to the cell distance.

**Comment by nahidrahmati** [comment]
Can we use it in HemoCellParticlefield.cpp? Actually I want to change the repulsive force method and there is two Cells that I want to calculate their Centers. should I iterate over infor_per_cell map? sonce I want only two cell, it may increase the calculation time.
